bundles/org.openhab.binding.siemensrds/src/main/java/org/openhab/binding/siemensrds/internal/RdsDataPoints.java

```diff
@@ -166,150 +158,99 @@ private void httpSetPointValueJson(String apiKey, String token, String pointId,
      */
-    private BasePoint getPoint(String hierarchyName) {
-        if (hierarchyName != null) {
-            for (Map.Entry<String, BasePoint> entry : points.entrySet()) {
-                BasePoint point = entry.getValue();
-                if (point != null && point.hierarchyName != null && point.hierarchyName.contains(hierarchyName)) {
-                    return point;
-                }
-            }
+    public BasePoint getPointByClass(String pointClass) throws RdsCloudException {
+        if (indexClassToId.isEmpty()) {
+            initClassToIdNameIndex();
         }
-        return null;
-    }
-
-    /*
-     * private method: retrieve the data point with the given hierarchyName
-     */
-    private String getPointId(String hierarchyName) {
-        if (hierarchyName != null) {
-            for (Map.Entry<String, BasePoint> entry : points.entrySet()) {
-                BasePoint point = entry.getValue();
-                if (point != null && point.hierarchyName != null && point.hierarchyName.contains(hierarchyName)) {
-                    return entry.getKey();
-                }
+        @Nullable
+        String pointId = indexClassToId.get(pointClass);
+        if (pointId instanceof String) {
+            BasePoint point = getPointById(pointId);
+            if (point instanceof BasePoint) {
+                return point;
             }
         }
-        return null;
+        throw new RdsCloudException(String.format("pointClass \"%s\" not found", pointClass));
     }
 
     /*
-     * public method: retrieve the state of the data point with the given
-     * hierarchyName
+     * public method: retrieve the data point with the given pointId
      */
-    public synchronized State getRaw(String hierarchyName) {
-        BasePoint point = getPoint(hierarchyName);
-        if (point != null) {
-            State state = point.getRaw();
-
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("getRaw: {}={}", hierarchyName, state.toString());
-            }
-            return state;
-        }
-        LOGGER.warn("getRaw: {}=No Value!", hierarchyName);
-        return null;
-    }
-
-    /*
-     * public method: return the presentPriority of the data point with the given
-     * hierarchyName
-     */
-    public synchronized int getPresPrio(String hierarchyName) {
-        BasePoint point = getPoint(hierarchyName);
-        if (point != null) {
-            int presentPriority = point.getPresentPriority();
-
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("getPresentPriority: {}={}", hierarchyName, presentPriority);
+    public BasePoint getPointById(String pointId) throws RdsCloudException {
+        Map<String, BasePoint> points = this.points;
+        if (points != null) {
+            @Nullable
+            BasePoint point = points.get(pointId);
+            if (point instanceof BasePoint) {
+                return point;
             }
```

Comment:
If you want another way around the null checker map issue, you can also do this:
```suggestion
            if (points.contains(pointId)) {
                return points.get(pointId);
            }
```

Reply:
> If you want another way around the null checker map issue

Personally I was quite happy using `instanceof` -- your suggestion to use contains() followed by get() would work, but it has a performance penalty, I think.

Reply:
There is a performance hit, but it would be minimal. Personally, I prefer just making the map value nullable. I was just showing here one of the many ways you can work around the null checker's poor handling of Map classes. 
But at the very least, please don't use instanceof  for your null checks. It makes the code very misleading to read.

Reply:
Done.